evm/utils/test_builder/normalization.py

```diff
@@ -0,0 +1,103 @@
+from collections.abc import (
+    Iterable,
+    Mapping,
+)
+
+from cytoolz import (
+    assoc_in,
+    compose,
+    identity,
+)
+from eth_utils import (
+    big_endian_to_int,
+    decode_hex,
+    is_0x_prefixed,
+    is_bytes,
+    is_hex,
+    to_canonical_address,
+)
+import cytoolz.curried
+import eth_utils.curried
+
+
+def state_list_to_dict(state_list):
+    """Convert state definitions in list form to the canonical dict form.
+
+    Example of state list elements:
+
+        - `(address, "balance", 1)`
+        - `(address, "nonce", 2)`
+        - `(address, "code", b"3")`
+        - `(address, "storage", 4, 5)`
+
+    For storage, the second to last entry specifies the slot and the last one the value.
+    """
+    d = {}
+    for state_item in state_list:
+        d = assoc_in(d, state_item[:-1], state_item[-1])
```

Comment:
It *might* be good to pre-validate that `state_list` is well formed.  Otherwise this is going to produce some very confusing results that might be hard to track down.  Similarly, enforcing uniqueness of all `state_item[:-1]` would fit into that validation to ensure that no duplicates are present.